t/unit/app/test_app.py

```diff
@@ -496,6 +496,16 @@ def foo():
         finally:
             _imports.MP_MAIN_FILE = None
 
+    def test_can_get_type_hints_for_tasks(self):
+        import typing
+
+        with self.Celery() as app:
+            @app.task
+            def foo(parameter: int) -> None:
+                pass
```

Comment:
This is invalid syntax in our Python 2.7 env. If necessary, perhaps we could make it use `exec()` like so:
```suggestion
            ctx = dict()
            exec("def foo(parameter: int) -> None: pass")
            foo = app.task(ctx["foo"])
```
Then you could mark this test with a skip when we're in a Python 2.